File: openapi-generator/php-laravel/lib/app/Models/DriverSafetyScoreResponse.php
<?php
/**
 * DriverSafetyScoreResponse
 */
namespace app\Models;

/**
 * DriverSafetyScoreResponse
 */
class DriverSafetyScoreResponse {

    /** @var int $crashCount Crash event count*/
    private $crashCount;

    /** @var int $driverId Driver ID*/
    private $driverId;

    /** @var int $harshAccelCount Harsh acceleration event count*/
    private $harshAccelCount;

    /** @var int $harshBrakingCount Harsh braking event count*/
    private $harshBrakingCount;

    /** @var \app\Models\SafetyReportHarshEvent[] $harshEvents */
    private $harshEvents;

    /** @var int $harshTurningCount Harsh turning event count*/
    private $harshTurningCount;

    /** @var int $safetyScore Safety Score*/
    private $safetyScore;

    /** @var string $safetyScoreRank Safety Score Rank*/
    private $safetyScoreRank;

    /** @var int $timeOverSpeedLimitMs Amount of time driven over the speed limit in milliseconds*/
    private $timeOverSpeedLimitMs;

    /** @var int $totalDistanceDrivenMeters Total distance driven in meters*/
    private $totalDistanceDrivenMeters;

    /** @var int $totalHarshEventCount Total harsh event count*/
    private $totalHarshEventCount;

    /** @var int $totalTimeDrivenMs Amount of time driven in milliseconds*/
    private $totalTimeDrivenMs;

}

File: openapi-generator/scala-finch/src/main/scala/org/openapitools/models/DriverSafetyScoreResponse.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import org.openapitools.models.SafetyReportHarshEvent
import scala.collection.immutable.Seq

/**
 * Safety score details for a driver
 * @param crashCount Crash event count
 * @param driverId Driver ID
 * @param harshAccelCount Harsh acceleration event count
 * @param harshBrakingCount Harsh braking event count
 * @param harshEvents 
 * @param harshTurningCount Harsh turning event count
 * @param safetyScore Safety Score
 * @param safetyScoreRank Safety Score Rank
 * @param timeOverSpeedLimitMs Amount of time driven over the speed limit in milliseconds
 * @param totalDistanceDrivenMeters Total distance driven in meters
 * @param totalHarshEventCount Total harsh event count
 * @param totalTimeDrivenMs Amount of time driven in milliseconds
 */
case class DriverSafetyScoreResponse(crashCount: Option[Int],
                driverId: Option[Int],
                harshAccelCount: Option[Int],
                harshBrakingCount: Option[Int],
                harshEvents: Option[Seq[SafetyReportHarshEvent]],
                harshTurningCount: Option[Int],
                safetyScore: Option[Int],
                safetyScoreRank: Option[String],
                timeOverSpeedLimitMs: Option[Int],
                totalDistanceDrivenMeters: Option[Int],
                totalHarshEventCount: Option[Int],
                totalTimeDrivenMs: Option[Int]
                )

object DriverSafetyScoreResponse {
    /**
     * Creates the codec for converting DriverSafetyScoreResponse from and to JSON.
     */
    implicit val decoder: Decoder[DriverSafetyScoreResponse] = deriveDecoder
    implicit val encoder: ObjectEncoder[DriverSafetyScoreResponse] = deriveEncoder
}
